bundles/org.openhab.core/src/main/java/org/openhab/core/internal/items/ExpireManager.java

```diff
@@ -244,13 +246,13 @@ public void receive(Event event) {
         }
         if (event instanceof ItemStateEvent) {
             ItemStateEvent isEvent = (ItemStateEvent) event;
-            processEvent(isEvent.getItemName(), isEvent.getItemState(), true);
+            processEvent(isEvent.getItemName(), isEvent.getItemState(), EventType.StateUpdate);
         } else if (event instanceof ItemCommandEvent) {
             ItemCommandEvent icEvent = (ItemCommandEvent) event;
-            processEvent(icEvent.getItemName(), icEvent.getItemCommand(), false);
+            processEvent(icEvent.getItemName(), icEvent.getItemCommand(), EventType.Command);
         } else if (event instanceof ItemStateChangedEvent) {
             ItemStateChangedEvent icEvent = (ItemStateChangedEvent) event;
-            processEvent(icEvent.getItemName(), icEvent.getItemState(), false);
+            processEvent(icEvent.getItemName(), icEvent.getItemState(), EventType.StateChanged);
         }
```

Comment:
Wouldn't it make sense to check for `ItemEvent` only and move this logic to `processEvent`? You could do something like
 
```
String itemName;
Type value;

if (event instance ItemStateEvent && !expireConfig.ignoreStateUpdates) {
    itemName = event.getName();
    type = ((ItemStateEvent) event).getItemState;
} else if (...) {
    ...
} else {
    return;
} 
```

The new enum could then be omitted. Looks cleaner to me.

Reply:
Essentially you are suggesting to merge `processEvent` and `receive`? Currently `processEvent` does not have access to the `event` object.

Reply:
Yes, currently the code just duplicates the check for the item type (one time in `receive` and one time in `processEvent`.

Reply:
Makes sense. See https://github.com/openhab/openhab-core/pull/2954/commits/fa9493626981d37a3eb7af6c6bac7782dfc1c2f4